llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -157,15 +157,29 @@ async def _task(
                     )
                 )
 
-    def send_event(self, message: Event) -> None:
+    def send_event(self, message: Event, step: Optional[str] = None) -> None:
         """Sends an event to a specific step in the workflow.
 
-        Currently we send all the events to all the receivers and we let
-        them discard events they don't want. This should be optimized so
-        that we efficiently send events where we know won't be discarded.
+        If step is None, the event is sent to all the receivers and we let
+        them discard events they don't want.
         """
-        for queue in self._queues.values():
-            queue.put_nowait(message)
+        if step is None:
+            for queue in self._queues.values():
+                queue.put_nowait(message)
+
```

Comment:
I think we should return early here, or put the rest in an else block, otherwise `step not in self._get_steps()` below would fail I think. I would also add a unit test to cover specifically when step is `None` (tests fail locally for me indeed).